processing/src/main/java/org/apache/druid/segment/IndexMergerV9.java

```diff
@@ -798,6 +799,41 @@ public File persist(
     );
   }
 
+  @Override
+  public File mergeSegmentFiles(
+      File[] segmentFiles,
+      boolean rollup,
+      AggregatorFactory[] metricAggs,
+      File outDir,
+      IndexSpec indexSpec,
+      @Nullable SegmentWriteOutMediumFactory segmentWriteOutMediumFactory
+  ) throws IOException
+  {
+    List<QueryableIndex> indexes = Lists.transform(Arrays.asList(segmentFiles), input -> {
+      try {
+        return indexIO.loadIndex(input);
+      }
+      catch (IOException e) {
+        Throwables.propagate(e);
```

Comment:
`Throwables.propagate()` must not be used without `throw ` before it. See https://github.com/google/guava/wiki/Why-we-deprecated-Throwables.propagate#propagate-is-magic. This method must not be used in new code. (Note: there are already 39 occurrences of this bug in Druid, #6701)